test/integration/job/job_test.go

```diff
@@ -1520,16 +1529,17 @@ func TestDelayTerminalPhaseCondition(t *testing.T) {
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.ElasticIndexedJob, true)
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobSuccessPolicy, test.enableJobSuccessPolicy)
 
-			closeFn, restConfig, clientSet, ns := setup(t, "delay-terminal-condition")
-			t.Cleanup(closeFn)
 			ctx, cancel := startJobControllerAndWaitForCaches(t, restConfig)
 			t.Cleanup(cancel)
 
+			test.job.Name = fmt.Sprintf("test-job-%d", i)
 			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &test.job)
 			if err != nil {
 				t.Fatalf("Error %q while creating the job %q", err, jobObj.Name)
 			}
-			t.Cleanup(func() { removePodsFinalizer(ctx, t, clientSet, ns.Name) })
+			t.Cleanup(func() {
+				removePodsFinalizersByName(ctx, t, clientSet, ns.Name, []string{"fake.example.com/blockDeletion", "batch.kubernetes.io/job-tracking"})
```

Comment:
nit: I would suggest to declare the custom finalizer in the test, say `blockDeletionFinalizerForTest`. For the Job finalizer we can use the API const `JobTrackingFinalizer`.